lib/widgets/pages/settings/settings_page.dart

```diff
@@ -24,159 +22,227 @@ import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:gap/gap.dart';
 
+/// Returns the given callback if the user data has been loaded, otherwise
+/// returns null.
+void Function()? _tappableIfUserLoaded(
+  BuildContext context,
+  void Function(BuildContext context, UserLoaded loadedState) callback,
+) {
+  return switch (context.read<UserCubit>().state) {
+    final UserLoaded loadedState => () => callback(context, loadedState),
+    _ => null,
+  };
+}
+
+// This callback is placed outside of any widgets because it is used by
+// multiple widgets.
+void _creditsTapCallback(BuildContext context) {
+  Navigator.push(context, CreditsPage.route).ignore();
+}
+
 class SettingsPage extends StatelessWidget {
-  const SettingsPage({required this.scrollController});
+  const SettingsPage._({required this.scrollController});
```

Comment:
Private constructor to convey that `SettingsPage.routeWith` should be used to navigate to this page.